### PR TITLE
validateXmlSvgRoot(): process % geometry

### PR DESCRIPTION
**Pull request by artificiel**
Affinity designer exports "flattened" SVG like this:
```xml
<svg width="100%" height="100%" viewBox="0 0 1920 1080" 
version="1.1" xmlns="http://www.w3.org/2000/svg" 
xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve">
```
this % approach throws of svgtiny and it generates all vertices at (0, 0). SVGloader currently checks for existence of wattr/hattr, but not if they're numbers, and apparently svgtiny does not process %.

the proposed addition fixes this by checking if width and height are set to a %-value. if so, it replaces the width and height with the viewBox-corresponding proportional values. 

in other words, it should not be a breaking change as 
 - an SVG that has width/height defined as values with not be affected (as currently);
 - an SVG with no width-height will be set to 100% view box (as currently);
 - an SVG with % would not work.

It emits warning if a SVG has neither view box or absolute width/height.

[edit: whitespace got a bit nuked in the method; check with "hide whitespace" enabled]

### CODE REVIEW

**Comment by Daandelange** [comment]
Hi, first thanks for this addon @NickHardeman ! I just tried it in oF 0.11.2 on osx 10.12 (max c++14), and there are some C++17 features introduced by this PR. Maybe it's worth tagging [the commit before: a4af835](https://github.com/NickHardeman/ofxSvgLoader/commit/a4af835fa77e95ddef9ad884367a0d29a90b0381) for max OF 0.11 compatibility (or c++14) ?

**Comment by artificiel** [comment]
ah well it only makes use of a `std::optional<glm::vec2> restore_to_viewbox` to track what values the viewport should be restored — it is easily adaptable by changing the optional to a vec2 + a bool (the pattern that optional strives to get rid of) and maintain 0.11 compatibility.

i am not in an OF dev environment for the current period but could do it later (unless someone else does it before),

**Comment by NickHardeman** [comment]
Hi @Daandelange, thank you for pointing this out. I just pushed some changes that should maintain compatibility for previous versions of OF ( via @artificiel's suggestions ). 
I also removed ofMain includes and added a 'getAllElementsForType' function to traverse the entire tree, 
ie. `auto paths = svg->getAllElementsForType<ofxSvgPath>();`

@artificiel thank you for making me aware of the std::optional, will be using that in other future projects ;)

**Comment by Daandelange** [comment]
Awesome, thanks, works great.

I just learned about `std::optional` too, not long after discovering the equivalent `Optional<Type>` in Rust ^_^
